gapis/shadertools/shadertools.go

```diff
@@ -189,6 +189,35 @@ func descriptorBindingLess(a DescriptorBinding, b DescriptorBinding) bool {
 	return a.SpirvId < b.SpirvId
 }
 
+func ParseSourceCode(shader []uint32) (string, error) {
+	spvReflectErr := func(res C.SpvReflectResult) error {
+		if res == C.SPV_REFLECT_RESULT_SUCCESS {
+			return nil
+		}
+		return fmt.Errorf("SPIRV-Reflect failed with error code %v\n", res)
+	}
+
+	module := C.SpvReflectShaderModule{}
+
+	shaderPtr := unsafe.Pointer(nil)
+	if len(shader) > 0 {
```

Comment:
Instead of relying on SPIRV-Reflect to deal with empty shaders, we should just return the empty string/an error here.